src/ml_board/frontend/Dashboard/src/worker_socket/MainThreadCallbacks.ts

```diff
@@ -38,24 +50,30 @@ export const updateMainThreadCallback = (parsedSocketData: DataFromSocket) => {
     MapEventToProcess[eventType as keyof typeof MapEventToProcess](dataToRedux, parsedSocketData.payload);
 
     // this is sent as a reply ONLY AFTER the first time
-    postMessage(dataToRedux);
+    bufferQueue.push(dataToRedux);
+
+    if (bufferQueue.length > 0  && (bufferQueue.length >= 1000 || bufferWindow === 0)) {
```

Comment:
at the first second of each Interval bufferWindow stays 0 for a whole second, this is why we get the rapid updates in the beginning. Then it updates later (suddenly) once we fill the whole `bufferQueue` (what seems to be the entire queue but i doubt it, because again `let data = d[0];`)